### DISCUSSION TITLE
270 Apache License Notices

### DISCUSSION THREAD

**Discussion by mstijak** [firebase/firebase-js-sdk#6669] [General]
I'm using Firebase in a fairly simple web application for authentication and document storage. 
The bundle size is pretty big - 800kB. After checking inside the files, I noticed that the Apache License Notice appears 270 times. That alone increases the file size 150kB. 
webpack preserves comments with the @license flag in production mode, but this feels utterly wrong.
Has anybody else encountered the same problem?

![image](https://user-images.githubusercontent.com/433394/194758572-77a55b69-1d9e-4d44-9123-9ffa92d2852a.png)

**Comment by hsubox76** [comment]
This seems to be a fairly common problem among libraries/apps that when source code is compiled from many files into one bundle, the license comment from each one of those files makes it into the final bundled file. I'll look around to see if there's a standard solution but if you know of one off the bat, let me know.

Otherwise at first glance, it looks like the simplest way is to use a terser plugin to strip the comments, and then add back one overall license comment at the top in some kind of post processing step. I think I've tried this before on the CDN bundles though and adding the license back at the end screws up the source map.

**Reply by hsubox76** [reply]
Actually I think we can get around this by using `rollup-plugin-license` to add an overarching license comment to the top during build time using Typescript's "pinned comment" format and then use Typescript's `removeComments` option to remove all non-pinned comments. Will look into it.

**Comment by r-thomson** [comment]
For anyone using esbuild (or Vite), setting `legal-comments` to `eof` seems to mostly de-dupe the licenses.

https://esbuild.github.io/api/#legal-comments

I was able to reduce my Firebase chunk from 217KB to 156KB with this setting.
